packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/label/LabelContext.ts

```diff
@@ -0,0 +1,16 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import React from 'react';
+import { Label } from '../DiagramRenderer.types';
+
+export const LabelContext = React.createContext<Label | undefined>(undefined);
```

Comment:
Just put an empty Label here and remove the `undefined` it will simplify tons of code below that should never be used if the label is not there. In case you need to explicitly handle the lack of value, prefer `null` next time to help distinguish values which do not exist.